### PR TITLE
Feature/global ens ee2 edits

### PR DESCRIPTION
**Pull request by StevenSimon-NOAA**
## Pull Request Testing ##

- [ ] Describe testing already performed for this Pull Request:</br>

Due to the ongoing slowness of Cactus and the recent timing out of global_ens jobs, preliminary testing has been limited thus far. The primary focus is confirming that the implemented EE2 global_ens edits that NCO prescribed run coherently with the EVS v1 global_ens component's workflow at all steps. Such edits include pushing all global_ens j-jobs up to the /jobs directory from the /jobs/component/step directories and adjusting all drivers accordingly, updating hardcoded paths in the global_ens module files, renaming select ush files (global_det to global_ens), eliminating unused global_ens ush files and commented out code where applicable, and removing export LD_LIBRARY_PATH from global_ens scripts and ush files.         

- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

Run all global_ens atmos and wave drivers for the duration in which we can confirm that all j-jobs have been properly initiated in the global_ens workflow at each step. Focus primarily on whether the plots drivers run to completion, as these are less likely to time out.   

- [ ] Has the code been checked to ensure that no errors occur during the execution? **[Yes or No]**

Yes, but in a limited fashion due to the slowness of Cactus. However, none of the implemented edits should introduce new features and thus new errors into the EVS global_ens workflow.  

- [ ] Do these updates/additions include sufficient testing updates? **[Yes or No]**

Yes.

- [ ] Please complete this pull request review by **[Fill in date]**.</br>

ASAP, but no later than October 13, 2023.

## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

Overview: This PR addresses and implements NCO's prescribed EE2 edits for the EVS v1 global_ens component following EC Flow testing and review. These edits include pushing all global_ens j-jobs up to the /jobs directory from the /jobs/component/step directories and adjusting all drivers accordingly, updating hardcoded paths in the global_ens module files, renaming select ush files (global_det to global_ens), eliminating unused global_ens ush files and commented out code where applicable, and removing export LD_LIBRARY_PATH from global_ens scripts and ush files. None of these edits are expected to introduce new features or errors into the EVS global_ens workflow.

### CODE REVIEW

**Comment by AliciaBentley-NOAA** [comment]
@StevenSimon-NOAA I looked through your PR and found that you successfully made all of the EE2 edits that Shelley assigned to you! I have accepted your checked boxes in the EE2 checklist document, grayed them out, and moved them down the list. Great job! 

@ShelleyMelchior-NOAA I support testing this global_ens EE2 PR whenever you are able to. Thanks!

**Comment by ShelleyMelchior-NOAA** [comment]
Excellent. I also did a cursory code review and all looks as expected. Me or Perry can begin this testing soon.

**Comment by ShelleyMelchior-NOAA** [comment]
Hi @StevenSimon-NOAA ! Lin needs walltimes increased for the following drivers. Can you please do the adjustments in this PR? Perry has already made the adjustments in the emc.vpppg parallel.
```
jevs_global_ens_atmos_gefs_snowfall_past31days_plots
#PBS -l walltime=01:00:00

jevs_global_ens_atmos_gefs_snowfall_past90days_plots
#PBS -l walltime=01:00:00

jevs_global_ens_ecme_atmos_grid2grid_stats
#PBS -l walltime=03:00:00

jevs_global_ens_gefs_atmos_grid2grid_stats
#PBS -l walltime=03:00:00
```

**Comment by StevenSimon-NOAA** [comment]
> Hi @StevenSimon-NOAA ! Lin needs walltimes increased for the following drivers. Can you please do the adjustments in this PR? Perry has already made the adjustments in the emc.vpppg parallel.
> 
> ```
> jevs_global_ens_atmos_gefs_snowfall_past31days_plots
> #PBS -l walltime=01:00:00
> 
> jevs_global_ens_atmos_gefs_snowfall_past90days_plots
> #PBS -l walltime=01:00:00
> 
> jevs_global_ens_ecme_atmos_grid2grid_stats
> #PBS -l walltime=03:00:00
> 
> jevs_global_ens_gefs_atmos_grid2grid_stats
> #PBS -l walltime=03:00:00
> ```

I updated the walltimes for these global_ens atmos plot and stats drivers and pushed them to the feature branch on my GitHub fork and this PR.

**Comment by malloryprow** [comment]
Did the fix for the creation of the /lfs
/h2/emc/vpppg/noscrub/emc.vpppg/evs/v1.0/prep/global_ens.YYYYmmdd get fixed in this?

**Comment by ShelleyMelchior-NOAA** [comment]
> Did the fix for the creation of the /lfs /h2/emc/vpppg/noscrub/emc.vpppg/evs/v1.0/prep/global_ens.YYYYmmdd get fixed in this?

No. Not addressed yet. There will need to be a follow-on global_ens PR which covers the higher hanging fruit.

**Comment by AliciaBentley-NOAA** [comment]
@StevenSimon-NOAA I see that you've already done this in at least some places, but just in case, please make sure that all of the global_ens scripts that previous called "global_det" files are updated to look for "global_ens" files now as well. 

For example, I see that you've already updated Line 685 in [ush/global_ens/evs_get_gens_atmos_data.sh](https://github.com/NOAA-EMC/EVS/pull/272/files#diff-13db435c6e9687549bffbc9208b764c545c0309a7f983bb6fa7f1f7237c9d3cfR685) to use "global_ens"

Apologies if you've already corrected this in all of the places! Thanks for your help with this!

**Comment by StevenSimon-NOAA** [comment]
> @StevenSimon-NOAA I see that you've already done this in at least some places, but just in case, please make sure that all of the global_ens scripts that previous called "global_det" files are updated to look for "global_ens" files now as well.
> 
> For example, I see that you've already updated Line 685 in [ush/global_ens/evs_get_gens_atmos_data.sh](https://github.com/NOAA-EMC/EVS/pull/272/files#diff-13db435c6e9687549bffbc9208b764c545c0309a7f983bb6fa7f1f7237c9d3cfR685) to use "global_ens"
> 
> Apologies if you've already corrected this in all of the places! Thanks for your help with this!

All applicable file renaming updates have been made in this PR for /ush/global_ens/.

**Comment by StevenSimon-NOAA** [comment]
> > Did the fix for the creation of the /lfs /h2/emc/vpppg/noscrub/emc.vpppg/evs/v1.0/prep/global_ens.YYYYmmdd get fixed in this?
> 
> No. Not addressed yet. There will need to be a follow-on global_ens PR which covers the higher hanging fruit.

This update has been made. Now testing.

**Comment by AliciaBentley-NOAA** [comment]
@StevenSimon-NOAA Great job with this! 

One quick thing: I think that the COMOUT paths in the wave J-jobs should use "compath.py **-o**" (where the "-o" is used because you are defining COM**OUT** and not COMIN). The paths after "compath.py -o" look good to me. @ShelleyMelchior-NOAA can confirm that you just need to add "-o".

Also, in order for "compath.py -o" to point the the emc.vpppg parallel and not production, you need to set COMROOT to the following in the global_ens wave dev drivers: 
export COMROOT=/lfs/h2/emc/vpppg/noscrub/$USER

Basically, "compath.py -o" takes whatever COMROOT is set to and appends the path after it. So if:
export COMROOT=/lfs/h2/emc/vpppg/noscrub/$USER
then
export COMOUT=${COMOUT:-$(compath.py -o ${NET}/${evs_ver})/${STEP}/${COMPONENT}/${RUN}}
results in 
COMOUT=/lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v1.0/prep/global_ens/wave

**Comment by StevenSimon-NOAA** [comment]
> @StevenSimon-NOAA Great job with this!
> 
> One quick thing: I think that the COMOUT paths in the wave J-jobs should use "compath.py **-o**" (where the "-o" is used because you are defining COM**OUT** and not COMIN). The paths after "compath.py -o" look good to me. @ShelleyMelchior-NOAA can confirm that you just need to add "-o".
> 
> Also, in order for "compath.py -o" to point the the emc.vpppg parallel and not production, you need to set COMROOT to the following in the global_ens wave dev drivers: export COMROOT=/lfs/h2/emc/vpppg/noscrub/$USER
> 
> Basically, "compath.py -o" takes whatever COMROOT is set to and appends the path after it. So if: export COMROOT=/lfs/h2/emc/vpppg/noscrub/$USER then export COMOUT=${COMOUT:-$(compath.py -o ${NET}/${evs_ver})/${STEP}/${COMPONENT}/${RUN}} results in COMOUT=/lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v1.0/prep/global_ens/wave

I will look into this tomorrow. I do not have my work laptop with me at home. 

The formatting for these COMOUT paths is currently inconsistent between atmos and wave components. None of the wave j-job COMOUTs feature the -o flag. And none of the j-jobs define COMROOT.

**Comment by AliciaBentley-NOAA** [comment]
@StevenSimon-NOAA Sure- more on this tomorrow when you get your laptop back in the morning! 

We haven't gotten to the global_det PR yet, so your global_ens PR will be corrected first. COMIN paths should use compath.py in the j-jobs and COMOUT paths should use compath.py -o in the J-jobs. 

COMROOT should be defined in the dev drivers, not the j-jobs. There were a lot of words in my last message. That point may have been lost. Thanks!

**Comment by ShelleyMelchior-NOAA** [comment]
@AliciaBentley-NOAA , more to come on the COMIN / COMOUT business. @StevenSimon-NOAA and I left it partially completed at the end of his work day. We will pick up on finalizing this, and other inappropriate usages for compath.py, tomorrow. Thanks for your watchful eye.

File: jobs/JEVS_GLOBAL_ENS_WAVE_GRID2OBS_PREP [hunk 1]
```diff
@@ -59,7 +59,7 @@ export COMIN=${COMIN:-$(compath.py ${envir}/com/${NET}/${evs_ver})}
 export COMINmodel=${COMINmodel:-$(compath.py ${envir}/com/${MODELNAME}/${gefs_ver})} 
 export COMINgdas=${COMINgdas:-$(compath.py ${envir}/com/obsproc/${obsproc_ver}/gdas)}
 
-export COMOUT=${COMOUT:-${COMIN}/${STEP}/${COMPONENT}/${RUN}}
+export COMOUT=${COMOUT:-$(compath.py ${NET}/${evs_ver})/${STEP}/${COMPONENT}/${RUN}}
```
Comment (ShelleyMelchior-NOAA) [inline jobs/JEVS_GLOBAL_ENS_WAVE_GRID2OBS_PREP]: As noted by Alicia, she beat me to the punch, these need `-o` added. I didn't get a chance to explain it in our little lesson earlier today. But please change line 62 to: 
```
export COMOUT=${COMOUT:-$(compath.py -o ${NET}/${evs_ver})/${STEP}/${COMPONENT}/${RUN}}
```
I can explain why in chat tomorrow.

File: jobs/JEVS_GLOBAL_ENS_WAVE_GRID2OBS_STATS [hunk 2]
```diff
@@ -60,7 +60,7 @@ export COMINmodel=${COMINmodel:-$(compath.py ${envir}/com/${MODELNAME}/${gefs_ve
 export ARCgefs=${COMIN}/prep/${COMPONENT}
 export COMINgdasnc=${COMIN}/prep/${COMPONENT}
 
-export COMOUT=${COMOUT:-${COMIN}/${STEP}/${COMPONENT}}
+export COMOUT=${COMOUT:-$(compath.py ${NET}/${evs_ver})/${STEP}/${COMPONENT}}
```
Comment (ShelleyMelchior-NOAA) [inline jobs/JEVS_GLOBAL_ENS_WAVE_GRID2OBS_STATS]: As noted by Alicia, she beat me to the punch, this needs `-o` added. I didn't get a chance to explain it in our little lesson earlier today. But please change line 63 to: 
```
export COMOUT=${COMOUT:-$(compath.py -o ${NET}/${evs_ver})/${STEP}/${COMPONENT}}
```
I can explain why in chat tomorrow.

File: jobs/JEVS_GLOBAL_ENS_PLOTS
Comment (ShelleyMelchior-NOAA) [inline jobs/JEVS_GLOBAL_ENS_PLOTS:1]: Line 30 needs to be changed to be:
```
export COMOUT=${COMOUT:-$(compath.py -o $NET/${evs_ver})}/$STEP/$COMPONENT/$RUN.$VDATE
```
Whenever the `-o` is used, you do not want to have `${envir}/com` in the path, no matter what $COMROOT is defined as.
Alicia's comment in the main conversation of this PR alluded to adding a COMROOT definition to the driver script. You can so this, but it isn't necessary since COMOUT is defined in the driver, which will override the default definition of COMOUT in the J-job. But if we were to define COMROOT in the driver, then we woudn't need to define COMOUT in the driver. It is confusing and we can talk about it more tomorrow.

File: jobs/JEVS_GLOBAL_ENS_PREP
Comment (ShelleyMelchior-NOAA) [inline jobs/JEVS_GLOBAL_ENS_PREP:1]: Line 52 needs to be changed to be:
```
export COMOUT=${COMOUT:-$(compath.py -o $NET/${evs_ver})}/$STEP/${COMPONENT}
```
Whenever the `-o` is used, you do not want to have `${envir}/com` in the path, no matter what $COMROOT is defined as.
Alicia's comment in the main conversation of this PR alluded to adding a COMROOT definition to the driver script. You can so this, but it isn't necessary since COMOUT is defined in the driver, which will override the default definition of COMOUT in the J-job. But if we were to define COMROOT in the driver, then we woudn't need to define COMOUT in the driver. It is confusing and we can talk about it more tomorrow.

File: jobs/JEVS_GLOBAL_ENS_STATS
Comment (ShelleyMelchior-NOAA) [inline jobs/JEVS_GLOBAL_ENS_STATS:1]: Line 34 needs to be changed to be:
```
export COMOUT=${COMOUT:-$(compath.py -o $NET/$evs_ver)}/$STEP/$COMPONENT
```
Whenever the `-o` is used, you do not want to have `${envir}/com` in the path, no matter what $COMROOT is defined as.
Alicia's comment in the main conversation of this PR alluded to adding a COMROOT definition to the driver script. You can so this, but it isn't necessary since COMOUT is defined in the driver, which will override the default definition of COMOUT in the J-job. But if we were to define COMROOT in the driver, then we woudn't need to define COMOUT in the driver. It is confusing and we can talk about it more tomorrow.

**Comment by StevenSimon-NOAA** [comment]
> @StevenSimon-NOAA Sure- more on this tomorrow when you get your laptop back in the morning!
> 
> We haven't gotten to the global_det PR yet, so your global_ens PR will be corrected first. COMIN paths should use compath.py in the j-jobs and COMOUT paths should use compath.py -o in the J-jobs.
> 
> COMROOT should be defined in the dev drivers, not the j-jobs. There were a lot of words in my last message. That point may have been lost. Thanks!

So it appears that the global_ens drivers do not currently feature a defined COMROOT env_var. All of the COMINs and COMOUTs in the drivers feature COMROOT as an explicit path /lfs/h2/emc/vpppg/noscrub/${USER}, not as an aliased env_var named COMROOT. Will I need to make such an edit to all of the global_ens driver scripts?

Thanks for the additional clarification regarding the compath.py -o edit in the j-jobs.

**Comment by ShelleyMelchior-NOAA** [comment]
As long as COMOUT is in the drivers, you do not need to define COMROOT in the drivers. But you can. It won't hurt anything. Look at this line:
```
export COMOUT=${COMOUT:-$(compath.py -o $NET/$evs_ver)}/$STEP/$COMPONENT
```
This is what is going on in it. It is saying, 
"Let's set COMOUT to something!"
"OK"
"If COMOUT has been previously defined in an upstream process (i.e., our dev drivers) `${COMOUT`, continue to use that definition. But if COMOUT has not been previously defined in an upstream process ( `:-$(compath.py -o $NET/$evs_ver)}` ), define COMOUT to be the result of the compath.py utility."

When we played w/ the compath.py utility in command line it used $COMROOT inherently b/c it is a global variable set when you load the prod_envir module.
```
$ module show prod_envir
------------------------------------------------------------------------------------------------------------
   /apps/ops/prod/nco/modulefiles/core/prod_envir/2.0.5.lua:
------------------------------------------------------------------------------------------------------------
setenv("OPSROOT","/lfs/h1/ops/prod")
setenv("OPSROOTssd","/lfs/f1/ops/prod")
setenv("COMROOT","/lfs/h1/ops/prod/com")
setenv("DATAROOT","/lfs/f1/ops/prod/tmp")
setenv("DCOMROOT","/lfs/h1/ops/prod/dcom")
setenv("PACKAGEROOT","/lfs/h1/ops/prod/packages")
setenv("DBNETROOT","/lfs/h1/ops/prod/dbnet")
```
This is where it gets confusing. For development purposes, one can redefine $COMROOT to be whatever they need it to be for their development work. 

So if we set COMROOT in our drivers to be `/lfs/h2/emc/vpppg/noscrub/emc.vpppg`, then we no longer need to first define COMOUT in the driver to have COMOUT predefined b/c then the compath.py utility will operate on the `$NET/$evs_ver` argument correctly. The reason it can't operate on it correctly right now is b/c evs is not yet installed on the operational system to be available in `/lfs/h1/ops/prod/com`.

Doing a quick search in there, you will not find us:
```
$ ls /lfs/h1/ops/prod/com
amsu_estimation  cpg          gfs      hurmsg    nam_mos        pcpanl           sfcship      uvi
aqm              dashboard    gfs_mos  hwm       narre          petss            smartinit    verf_g2g
baseline         date         glmp     hwrf      navgem         prod             spa_on_call  verf_gridtobs
blend            ecflow       glofs    hysplit   nesdis         psurge           spc_post     verf_precip
cadb             ecmwf        glwu     ifi       nhcg           radarl2          spcsref      wave_nfcens
ccpa             ecmwf_mos    godas    jma       nldas          rap              sref         wfs
cdas             ens_tracker  gpspw    lmp       nos            rcdas            stofs        wsa_enlil
cdas2            epoch        hafs     logs      nosofs         rtdm             swmf         wsp
cfs              estofs       hiresw   madis     nsst           rtma             thanks       wsr
cfs_hur          etss         hmon     mag       nwm            rtofs            tmp
cip              fnmoc_wave   href     marine34  nwps           runhistory       transfers
cmc              gefs         hrrr     naefs     obsproc        seaice_analysis  ukmet
cpci             gentracks    hry_mos  nam       parallel-test  seaice_drift     urma
```
But when you search on our newly defined COMROOT, you will find evs:
```
$ ls /lfs/h2/emc/vpppg/noscrub/emc.vpppg
evs  EVS  evs_driver  EVS_old_para  EVS_para_cactus_09.11.23  EVS_para_dogwood_09.11.23  EVS_save  verification
```
BOTTOMLINE ... your drivers are fine w/o defining COMROOT as long as COMOUT is defined. You just can't have the -o flag paired w/ `$envir/com` no  matter how or where COMROOT is defined b/c when NCO runs this in operations, they won't be predefining COMOUT in any upstream process.

File: jobs/JEVS_GLOBAL_ENS_WAVE_GRID2OBS_PLOTS [hunk 3]
```diff
@@ -56,7 +56,7 @@ export VERIF_CASE=${VERIF_CASE:-grid2obs}
 # Define COMIN/COMOUT variables     
 ####################################
 export COMIN=${COMIN:-$(compath.py ${envir}/com/${NET}/${evs_ver})}
-export COMOUT=${COMOUT:-$(compath.py ${envir}/com/${NET}/${evs_ver})}
+export COMOUT=${COMOUT:-$(compath.py -o $NET/${evs_ver})}/$STEP/$COMPONENT/$RUN.$VDATE
```
Comment (ShelleyMelchior-NOAA) [inline jobs/JEVS_GLOBAL_ENS_WAVE_GRID2OBS_PLOTS]: @StevenSimon-NOAA , can you check this one? Why did `$STEP/$COMPONENT/$RUN.$VDATE` get added to the end of this definition of COMOUT? This could be a problem b/c in the driver upstream of this code, `dev/drivers/scripts/global_ens/plots/jevs_global_ens_wave_grid2obs_plots.sh` , the COMOUT definition ends at ${evs_ver}. See line 51:
```
export COMOUT=${OUTPUTROOT}/${NET}/${evs_ver}
```

Comment (StevenSimon-NOAA) [reply jobs/JEVS_GLOBAL_ENS_WAVE_GRID2OBS_PLOTS]: I added it because I thought it was consistent with the formatting of atmos plot j-jobs. I now see that this is not the case for the driver upstream.

**Comment by ShelleyMelchior-NOAA** [comment]
Somehow KEEPDATA=YES did not happen despite it being checked off in the master checklist. @StevenSimon-NOAA , please update every driver script in `feature/global_ens-ee2-edits` to set KEEPDATA to YES.

**Comment by AliciaBentley-NOAA** [comment]
@StevenSimon-NOAA I have gone over the global_ens PR again and compared it to the remaining items that were checked off on the EE2 checklist. Everything is there and looks fantastic! I think this PR is ready for testing, @ShelleyMelchior-NOAA!

**Comment by ShelleyMelchior-NOAA** [comment]
Prep processing is underway. Some jobs have completed. Please review the output as the final job progresses.
```
83480797.cbqs01                shelley.melchi* dev             jevs_global_en*   167868    2   128  1000g 06:00 R 02:04
```
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/prep/global_ens`
logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_*prep.o8348*`

UPDATE: prep processing is complete!

**Comment by ShelleyMelchior-NOAA** [comment]
stats processing is underway for cmce. So far precip and snowfall have concluded.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/cmce.20231002`
logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_cmce_atmos_*stats.o83490*`

UPDATE: cmce stats processing is complete

**Comment by ShelleyMelchior-NOAA** [comment]
stats processing is complete for ecme
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/ecme.20231002`
logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_ecme_atmos_*stats.o83491*`

**Comment by ShelleyMelchior-NOAA** [comment]
stats processing is complete for naefs
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/naefs.20231002`
logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_naefs_atmos_*stats.o83491*`

**Comment by ShelleyMelchior-NOAA** [comment]
stats processing is complete for wave
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/wave.20231003`
logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_wave_grid2obs_stats.o83491904`

**Comment by ShelleyMelchior-NOAA** [comment]
stats processing is underway for gefs. As of this comment some processes have completed (sea ice, sst, and headline).
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/gefs.20231002`
logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_gefs_atmos_*stats.o835038*`

UPDATE: gefs stats processing is complete!

**Comment by ShelleyMelchior-NOAA** [comment]
stats processing is complete for gfs headline
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/gfs.20231002`
logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_gfs_headline_grid2grid_stats.o83504737`

**Comment by ShelleyMelchior-NOAA** [comment]
Final update for today ... all prep and stats processing has completed. Please review.

File: scripts/global_ens/stats/exevs_global_ens_wave_grid2obs_stats.sh
Comment (ShelleyMelchior-NOAA) [inline scripts/global_ens/stats/exevs_global_ens_wave_grid2obs_stats.sh:1]: @StevenSimon-NOAA , modify line 119 to look like this:
```
echo "export err=\$?; err_chk" >> ${DATA}/jobs/run_${MODELNAME}_${RUN}_${VDATE}${cyc2}_f${flead}_g2o.sh
```
The difference is the `\` before the `$?` when exporting the `err` variable.

**Comment by ShelleyMelchior-NOAA** [comment]
I have begun some plots processing. 

naefs has completed.
COMOUT=`/lfs/h2/emc/ptmp/shelley.melchior/evs/v1.0/plots/global_ens/atmos.20231003`
logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_atmos_naefs_*plots.o835593*`

wave has completed.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/plots/global_ens/wave.20231004`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_wave_grid2obs_plots.o83558968`

**Comment by StevenSimon-NOAA** [comment]
> Final update for today ... all prep and stats processing has completed. Please review.

All global_ens prep files (cmce, cmce_bc ecme, gefs, gefs_bc, osi, headline, wave) are accounted for by quantity for all cases. The full allotment of global_ens stats files are accounted for by quantity for all cases except small stats files for ecme snowfall, gefs sea ice, gefs sst, naefs g2g, and naefs g2o. Some of this may be due to the naefs v6 to v7 transition. Other discrepancies may be due to the some small gaps in the Parallel global_ens prep archive due to timing out of prep jobs, a result of recent slowness on Cactus.

**Comment by StevenSimon-NOAA** [comment]
> I have begun some plots processing.
> 
> naefs has completed. COMOUT=`/lfs/h2/emc/ptmp/shelley.melchior/evs/v1.0/plots/global_ens/atmos.20231003` logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_atmos_naefs_*plots.o835593*`
> 
> wave has completed. COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/plots/global_ens/wave.20231004` logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_wave_grid2obs_plots.o83558968`

All naefs and gefs wave plots are fully accounted for.

**Comment by ShelleyMelchior-NOAA** [comment]
I have started the last of the plots processing for all remaining drivers: `jevs_global_ens_atmos_gefs_*`

COMOUT=`/lfs/h2/emc/ptmp/shelley.melchior/evs/v1.0/plots/global_ens/atmos.20231003`
logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_atmos_gefs_*plots.o83566*`

**Comment by StevenSimon-NOAA** [comment]
> I have started the last of the plots processing for all remaining drivers: `jevs_global_ens_atmos_gefs_*`
> 
> COMOUT=`/lfs/h2/emc/ptmp/shelley.melchior/evs/v1.0/plots/global_ens/atmos.20231003` logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_atmos_gefs_*plots.o83566*`

All global_ens plots are fully accounted for by quantity.

**Comment by ShelleyMelchior-NOAA** [comment]
This PR is ready for merge after code review (been doing this along the way, so this is just a merge formality) and final sync fork.

**Review by ShelleyMelchior-NOAA — APPROVED** [review]
All testing produces expected results. This PR is ready for merge.
